docs/frontend.asciidoc

```diff
@@ -0,0 +1,108 @@
+[[frontend]]
+== Frontend support
+
+NOTE: Frontend support is an experimental feature not intended for production usage at the moment. 
+
+This section describes features specifically designed to support frontend application monitoring
+and a guide on how to enable experimental frontend support. 
+
+[[frontend-enable]]
+[float]
+=== Enable Frontend Support
+To try out experimental frontend support, set the `apm-server.frontend.enabled` to `true`.
+See https://github.com/elastic/apm-server/blob/{doc-branch}/apm-server.reference.yml[`apm-server.reference.yml`] for more configuration options.
+
+[[sourcemap]]
+=== Source Maps
+It is common practice to minify client-side JavaScript code for several reasons, e.g. performance gain. 
+This can make debugging very difficult, as it is hard to read the minified files.
+Source mapping can help for debugging minfied JavaScript files, 
+by mapping code from the minified files to the original source code. 
+
+APM Server has a source map <<sourcemap-endpoint,upload endpoint>> 
+which accepts source maps complying to the 
+https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k[Source map revision 3 proposal].
```

Comment:
alternative link in case HTML preference: http://sourcemaps.info/spec.html

Reply:
I didn't link this on purpose as the site says `Note! This is a copy of the standard. Any discrepencies should be resolved in favour of the Google Doc.`